src/pip/_internal/vcs/git.py

```diff
@@ -131,9 +131,15 @@ def get_revision_sha(cls, dest, rev):
             on_returncode='ignore',
         )
         refs = {}
-        for line in output.strip().splitlines():
+        # NOTE: We do not use splitlines here since that would split on other
+        #       unicode separators, which can be maliciously used to install a
+        #       different revision.
+        for line in output.strip().split("\n"):
+            line = line.rstrip("\r")
```

Comment:
This line is here because Windows. 😅

Reply:
why not `line = line.strip()` ?

Reply:
ah yes, there could be significant whitespace in the ref.

Reply:
Yup!

And, I think it's not likely that someone ends their refs with "\r" -- and if they did, I'm pretty sure things were broken anyway?